etcd/src/main/java/org/jclouds/etcd/domain/members/CreateMember.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.etcd.domain.members;
+
+import java.util.List;
+
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+@AutoValue
+public abstract class CreateMember {
+
+   @Nullable
+   public abstract String name();
+
+   public abstract List<String> peerURLs();
+
+   public abstract List<String> clientURLs();
+
+   CreateMember() {
+   }
+
+   @SerializedNames({ "name", "peerURLs", "clientURLs" })
+   public static CreateMember create(final String name, final List<String> peerURLs, List<String> clientURLs) {
```

Comment:
@zack-shoylev I think you made the point previously, maybe here or on IRC, that I agree with in that builders, at least so far as AutoValue classes are concerned, should be used when those classes are client facing. In this case the CreateMember class would be just that and so I think it makes sense to Builderize (?!?) it. Other classes which are not client facing, and are simply handed back from a function call, need not have this functionality. Maybe this is obvious but I thought it a point worth making. I'm actually writing up the builder class now so if we are in agreement I can send in another squashed commit with the builder functionality added on top. Going forward I will follow the same model with this provider, that being: client facing value classes will get builders if only to keep things consistent.